packages/components/menu/src/menu.ts

```diff
@@ -236,6 +239,15 @@ export default defineComponent({
       }
     )
 
+    watch(
```

Comment:
We can use `watchEffect` here, since we should watch `props.mode` too.

Reply:
You're right, I missed it.